.github/workflows/ci.yml

```diff
@@ -100,41 +105,3 @@ jobs:
         fail_ci_if_error: false
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-
-  documentation:
-
-    name: Documentation, Python 3.12, Ubuntu
-    runs-on: ubuntu-latest
-
-    steps:
-
-    - name: Checkout code
-      uses: actions/checkout@v4
-      with:
-        fetch-depth: 0
-
-    - name: Install graphviz and pandoc
-      run: sudo apt-get install graphviz pandoc
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: '3.12'
-        cache: pip
-
-    - name: Install nox uv
-      run: python -m pip install --progress-bar off --upgrade nox uv
-
-    - name: Cache
-      uses: actions/cache@v4
-      with:
-        path: |
-          .nox
-        key: docs-${{ runner.os }}-${{ hashFiles('requirements.txt') }}
-
-    - name: Build documentation with Sphinx
-      run: nox -e docs -- -q
-
-    - name: Print troubleshooting information on failure
-      if: ${{ failure() }}
-      run: echo -e $DOC_TROUBLESHOOTING
```

Comment:
There was a lot of duplication in the workflows for building docs and running tests, so I merged them together above.